### PR TITLE
Implement GetHeader and GetHeaders

### PR DESCRIPTION
**Pull request by xdsxc**
- Implement GetHeader and GetHeaders
- Update yaml pkg

### CODE REVIEW

**Review by qngn — APPROVED** [review]
Looks good to me

File: gmime/gmime_test.go [hunk 1]
```diff
@@ -671,6 +672,103 @@ func TestPart_String(t *testing.T) {
...
+	}
+	tcs := []TestCase{
+		{
+			filename: `test_data/mime_simple.eml`,
+			expected: []map[string]string{
+				{
+					"Content-Type": "text/html",
+				},
+				{
+					"Content-Transfer-Encoding": "quoted-printable",
```
Comment (ericychoi) [inline gmime/gmime_test.go:688]: can we have a test payload with the infamous `=3D` line delimiters?

File: gmime/part.go [hunk 2]
```diff
@@ -151,9 +151,32 @@ func (p *Part) SetHeader(name string, value string) {
 	C.g_mime_header_list_set(headers, cName, cValue, cCharset)
 }
 
-// Headers gives you all headers for part
-func (p *Part) Headers() textproto.MIMEHeader {
-	return nil
+// GetHeaders gives you all headers for part
+func (p *Part) GetHeaders() textproto.MIMEHeader {
```
Comment (ericychoi) [inline gmime/part.go:155]: usually you just don't name things `Get...()` in go

File: gmime/gmime_test.go (hunk 1 above)
Comment (xdsxc) [reply gmime/gmime_test.go:688]: That isn't going to get parsed here, this is only to extract the headers but not to decode quoted printable. You'll see that test on the next PR :)

File: gmime/part.go (hunk 2 above)
Comment (xdsxc) [reply gmime/part.go:155]: I figured it plays nicely with GetHeader & GetHeaders 🤷

**Pull request merged**